### PR TITLE
Add services to odo app describe

### PR DESCRIPTION
**Pull request by geoand**
## What is the purpose of this change? What does it change?
See $SUBJECT

## Was the change discussed in an issue?
Fixes: #1203

## How to test changes?
Create various component and/or services. 
Make sure that `odo app describe` shows the appropriate output

### CODE REVIEW

File: tests/e2e/link_test.go [hunk 1]
```diff
@@ -63,6 +63,13 @@ var _ = Describe("odoLinkE2e", func() {
 			})
 		})
 
+		It("app describe should show the mysql service", func() {
```
Comment (amitkrout) [inline tests/e2e/link_test.go]: LGTM. UTs won’t be a good idea here. Only user facing scenario (Integration test) would be enough jot verify a description change. 

However i want to test the pr just to see the output format

**Comment by geoand** [comment]
> @amitkrout That is a very good catch!
> 
> I don't know what the actual problem is, but it's unrelated to this PR - you should see the same output if you test `master`.
> I propose that you open a new issue, assign it to me and I will check it out as part of a new PR. Does that sound good?

Actually @amitkrout , hold on a sec, let check something

**Comment by geoand** [comment]
> > Actually @amitkrout , hold on a sec, let check something
> 
> Sure

Can you post the steps you followed to reproduce the problem?
I can't seem to reproduce it either with the PR or with master

**Comment by amitkrout** [comment]
@geoand Steps to reproduce https://github.com/redhat-developer/odo/pull/1252#pullrequestreview-198048365

1. Download minishift v1.30.1
2. start minishift
```
$ MINISHIFT_ENABLE_EXPERIMENTAL=y ./minishift start --extra-clusterup-flags "--enable=*,service-catalog,automation-service-broker,template-service-broker"
```
3. Fetch the pr and run make bin
4. ```./odo create nodejs frontend```
5. ```./odo create python backend```
6. ```./odo link backend --component frontend```
7. ```./odo service create mysql-persistent```
8. ```./odo link mysql-persistent -w --component backend```
9. ```./minishift console```
10. Create the external route for both ```frontend``` and ```backend```
11. ```./odo app describe```

Please correct me if i diverted the usecase in a wrong way

**Comment by geoand** [comment]
> @geoand Steps to reproduce [#1252 (review)](https://github.com/redhat-developer/odo/pull/1252#pullrequestreview-198048365) comment
> 
> 1. Download minishift v1.30.0
> 2. start minishift
> 
> ```
> $ MINISHIFT_ENABLE_EXPERIMENTAL=y ./minishift start --extra-clusterup-flags "--enable=*,service-catalog,automation-service-broker,template-service-broker"
> ```
> 1. Fetch the pr and run make bin
> 2. `./odo create nodejs frontend`
> 3. `./odo create python backend`
> 4. `./odo link backend --component frontend`
> 5. `./odo service create mysql-persistent`
> 6. `./odo link mysql-persistent -w --component backend`
> 7. `./minishift console`
> 8. Create the external route for both `frontend` and `backend`
> 9. `./odo app describe`
> 
> Please correct me if i diverted the usecase in a wrong way

OK, so I see what the problem is.
You need to create the URLs using  `odo url create` instead of doing it from the console.
Can you please check?

**Comment by amitkrout** [comment]
> You need to create the URLs using odo url create instead of doing it from the console.
Can you please check?

Right, This is the issue. 
However manually i created external route to use port 8080 and i think this should be captured by ```./odo app describe```

screen shot 2019-01-30 at 7 17 18 pm

**Comment by geoand** [comment]
> > You need to create the URLs using odo url create instead of doing it from the console.
> > Can you please check?
> 
> Right, This is the issue.
> However manually i created external route to use port 8080 and i think this should be captured by `./odo app describe`
> 
> screen shot 2019-01-30 at 7 17 18 pm

That makes sense (although I don't it's easy since we use specific labels that aren't added by the console).
Can you open another issue for this and ping Jorge to get his opionion?

Thanks @amitkrout

**Comment by amitkrout** [comment]
> Can you open another issue for this and ping Jorge to get his opionion?

Sure, will do that

**Comment by geoand** [comment]
> LGTM
> Will discuss the observation [#1252 (comment)](https://github.com/redhat-developer/odo/pull/1252#issuecomment-458950426) in a separate issue.

Thanks!

**Comment by kadel** [comment]
> @geoand it looks like that the `0` is shown in case of nodejs because there is not external port mapped. So for that we shouldn't show `exposed via 0` to the user. I would suggest to resolve that in this PR itself if that is OK. This could be confusing for the user.

If I understand this correctly, this issue is in the current master, this is not something that was introduced by this PR. If that is true then it shouldn't be resolved in this PR. Let's open a new issue to report this as a bug, and we can take care of it in separate PR.

**Comment by geoand** [comment]
> > @geoand it looks like that the `0` is shown in case of nodejs because there is not external port mapped. So for that we shouldn't show `exposed via 0` to the user. I would suggest to resolve that in this PR itself if that is OK. This could be confusing for the user.
> 
> If I understand this correctly, this issue is in the current master, this is not something that was introduced by this PR. If that is true then it shouldn't be resolved in this PR. Let's open a new issue to report this as a bug, and we can take care of it in separate PR.

This only happens when the URL is created **outside** of odo, so it has nothing to do with PR and I would argue that it probably doesn't even need to be handled by odo in general

**Comment by metacosm** [comment]
I move that we should merge this PR, then.

**Comment by kadel** [comment]
> This only happens when the URL is created **outside** of odo, so it has nothing to do with PR and I would argue that it probably doesn't even need to be handled by odo in general

I think that we can detect port from Route even if it is not created by odo. But it is not a priority right now. This is why I would like to create an Issue for it and then we can address this later, or decide that we don't want to do that. 

Let's discuss this elsewhere.

**Comment by geoand** [comment]
> > This only happens when the URL is created **outside** of odo, so it has nothing to do with PR and I would argue that it probably doesn't even need to be handled by odo in general
> 
> I think that we can detect port from Route even if it is not created by odo. But it is not a priority right now. This is why I would like to create an Issue for it and then we can address this later, or decide that we don't want to do that.
> 
> Let's discuss this elsewhere.

@amitkrout Have you already created the issue?

**Comment by amitkrout** [comment]
> @amitkrout Have you already created the issue?

Not yet, busy with some other stuff. @geoand Please create the issue, Later i will put the test content if needed

**Comment by girishramnani** [comment]
ok LGTM

**Comment by amitkrout** [comment]
@geoand Created https://github.com/redhat-developer/odo/issues/1256

**Comment by geoand** [comment]
Thanks @amitkrout ! Was going to do that in a while

**Pull request merged**